The fact that an enumeration type is actually an enumeration should not be duplicated in its name.


== Noncompliant Code Example

----
Enum FooEnum ' Noncompliant
    Foo
End Enum
----


== Compliant Solution

----
Enum Foo
    Foo
End Enum
----

