include::../description.adoc[]

== Noncompliant Code Example

----
Rem FIXME: Add documentation - Noncompliant
Function Divide(ByVal Numerator As Integer, ByVal Denominator As Integer) As Integer
    Divide = Numerator / Denominator ' FIXME: Denominator value might be 0 - Noncompliant
End Function
----

include::../see.adoc[]

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
