include::../description.adoc[]

== Noncompliant Code Example

----
For index As Integer = 6 To 5  // Noncompliant 
    // ...
Next
----

== Compliant Solution

----
For index As Integer = 0 To 5
    // ...
Next
----

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
