include::../description.adoc[]

== Noncompliant Code Example

----
For Index As Integer = 1 To 42 ' Empty on purpose or missing piece of code?
Next
----

== Compliant Solution

----
For Index As Integer = 1 To 42
    Process(Index)
Next
----

include::../exceptions.adoc[]
