include::../description.adoc[]

== Noncompliant Code Example

----
IF expression THEN
  RETURN TRUE;
ELSE
  RETURN FALSE;
END IF;
----

== Compliant Solution

----
RETURN expression;
----

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
