{cpp}17 version of the standards introduces ``++if constexpr++``. If the ``++constexpr++`` keyword follows the ``++if++`` keyword in an if statement, then the ``++if++`` condition must be a constant and the ``++then++`` or ``++else++`` block is discarded at compile time, depending on the value of the constant.


More precisely, if the condition of an if constexpr statement in a template is not value-dependent after instantiation, the discarded statement is not instantiated. This behaviour enables to write templates in a more readable way, without using complex patterns (eg: by using ``++std::enable_if++``) to make code compile. 


This rule points out where such patterns could simply be replaced by ``++if constexpr++``.


== Noncompliant Code Example

----

----


== Compliant Solution

----

----

