== Noncompliant Code Example

----
template <class T, int Size> 
class MyVector {
  MyVector() {
    if (Size > 42) { // Noncompliant, "if" can be replaced by "if constexpr"
      // small-size optimized
    } else {
      // general implementation
    }
  }

  // ...
};
----


== Compliant Solution

----
template <class T, int Size> 
class MyVector {
  MyVector() {
    if constexpr (Size > 42) {
      // small-size optimized
    } else {
      // general implementation
    }
  }

  // ...
};
----

