=== Relates: RSPEC-949

=== On 2014-10-16T13:02:35Z Ann Campbell Wrote:
\[~samuel.mercier] please:

* fill in the appropriate reference field(s).
* provide a See section.
* use double curly braces around code in the description, e.g. NULL.
* use the standard section titles: Noncompliant Code Example, Exception*s*

=== On 2014-10-17T15:42:00Z Ann Campbell Wrote:
\[~freddy.mallet] this rule an RSPEC-949 are 2 sides of the same coin

