Creating temporary primitive wrapper objects only for ``++String++`` conversion or the use of the ``++compareTo++`` method is inefficient.

Instead, the static ``++toString()++`` or ``++compare++`` method of the primitive wrapper class should be used.

== Noncompliant Code Example

----
new Integer(myInteger).toString();  // Noncompliant
----

== Compliant Solution

----
Integer.toString(myInteger);        // Compliant
----
