include::../description.adoc[]

== Noncompliant Code Example

----
int GetBestNumber()
{
  return 12;  // Noncompliant
}
----

== Compliant Solution

----
const int BestNumber = 12;
----
or

----
static readonly int BestNumber = 12;
----

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
