Titles are important because they are displayed in search engine results as well as the browser's toolbar.


This rule verifies that the ``++<head>++`` tag contains a ``++<title>++`` one, and the ``++<html>++`` tag a ``++<head>++`` one.


== Noncompliant Code Example

----
&lt;html&gt;         &lt;!-- Non-Compliant --&gt;

&lt;body&gt;
...
&lt;/body&gt;

&lt;/html&gt;
----


== Compliant Solution

----
&lt;html&gt;         &lt;!-- Compliant --&gt;

&lt;head&gt;
  &lt;title&gt;Some relevant title&lt;/title&gt;
&lt;/head&gt;

&lt;body&gt;
...
&lt;/body&gt;

&lt;/html&gt;
----


