The `+LONG+` and `+LONG RAW+` datatypes are deprecated and Oracle recommends to migrate them to the LOB datatypes `+CLOB+`, `+NCLOB+` or `+BLOB+`.


== Noncompliant Code Example

----
CREATE TABLE images(
  data LONG RAW
);
----


== Compliant Solution

----
CREATE TABLE images(
  data BLOB
);
----


