include::../description.adoc[]

== Noncompliant Code Example

----
strlen($name); // Noncompliant; "strlen" has no side effect
----

== Compliant Solution

----
$length = strlen($name);
----

include::../see.adoc[]
