include::../description.adoc[]

== Noncompliant Code Example

----
protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
{
  optionsBuilder.UseSqlServer("Server=myServerAddress;Database=myDataBase;User Id=myUsername;Password="); // Noncompliant
}
----

In https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/[appsettings.json]

----
{
  "ConnectionStrings": {
    "DefaultConnection": "Server=myServerAddress;Database=myDataBase;User Id=myUsername;Password="
  }
}
----

== Compliant Solution

----
protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
{
  optionsBuilder.UseSqlServer("Server=myServerAddress;Database=myDataBase;Integrated Security=True");
}
----

In https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/[appsettings.json]

----
{
  "ConnectionStrings": {
    "DefaultConnection": "Server=myServerAddress;Database=myDataBase;Integrated Security=True"
  }
}
----

include::../see.adoc[]
