The transitive property says that if ``++a == b++`` and ``++b == {cpp}``, then ``++a == {cpp}``. In such cases, there's no point in assigning ``++a++`` to ``++{cpp}`` or vice versa because they're already equivalent. 

This rule raises an issue when an assignment is useless because the assigned-to variable already holds the value on all execution paths.


== Noncompliant Code Example

----
a = b;
c = a;
b = c; // Noncompliant: c and b are already the same
----


== Compliant Solution

----
a = b;
c = a;
----

