{cpp}14 has introduced transparent comparators: the function objects that support heterogeneous comparison (i.e., comparison of values of different types, such as ``std::string`` and ``char const*``). When using such comparator, the search-optimized containers, namely, ``std::set``, ``std::multiset``, ``std::map``, and ``std::multimap``, enable additional lookup-function overloads that support types different from the ``key_type``.

Invoking a lookup function (such as ``find``, ``count``, or ``lower_bound``) with a non-``std::string`` argument, i.e., a raw C-string literal (``s.find("Nemo")``), or a temporary ``std::string`` created of an ``std::string_view``, on a container of ``std::string`` with non-transparent comparator, leads to a temporary ``std::string`` object, because the lookup function will support only an argument of the ``key_type``.

Always prefer using a transparent comparator with associative "std::string" containers to avoid creating the temporary. Note that transparent comparators are strongly discouraged if used with types that are not directly comparable as it will lead to the creation of ``O(log(container.size())))`` temporaries with lookup functions such as ``find``, ``count``, and ``lower_bound``.

This rule will detect ``std::set``, ``std::multiset``, ``std::map``, and ``std::multimap`` types with a non-transparent comparator.


== Noncompliant Code Example

----
void f() {
  // the default std::less<std::string> is not transparent
  std::set<std::string> m = { "Dory", "Marlin", "Nemo", "Emo"}; // Noncompliant
  m.find("Nemo"); // This leads to a temporary std::string{"Nemo"}.
  std::string_view n{"Nemo"};
  m.find(std::string(n)); // extra temporary std::string
}
----


== Compliant Solution

----
void f() {
  // std::less<> is transparent
  std::set<std::string, std::less<>> m = // Compliant
      { "Dory", "Marlin", "Nemo", "Emo"};
  m.find("Nemo"); // No temporary is created, the raw C-string literal
                  // is compared directly with std::string elements
  std::string_view n{"Nemo"};
  m.find(n); // No need to create the std::string 
}
----

