Heterogeneous containers were introduced in {cpp}14 to increase performances when working with ``++std::map++``, ``++std::set++``, ``++std::multimap++`` and ``++std::multiset++``. Before their introduction, when working with a key in such a container, it was required to use an object of the exact key type. This could lead to costly object creations when we would like to work with object of compatible but different types:

----
std::map<std::string, int> m;
m.find("abc"); // Will convert the char const * to a std::string, maybe performing costly memory allocation
----

With heterogeneous containers, the previous code will not create a ``++string++``, but directly compare keys in the map with the searched ``++char const *++``. This is a behaviour you can opt-in by defining as a comparison method for the map one which is called _transparent_.

----
std::map<std::string, int, std::less<>> m; // std::less<> is transparent, std::less<anything> is not
m.find("abc"); // Will directly compare the char const* with the keys, no object conversion
----

A transparent comparator is one that declares a nested type named ``++is_transparent++``. It is supposed to be able to compare heterogeneous object types, for instance in this case compare a ``++char const *++`` with a ``++string++``, without performing any conversion.

What happens if it cannot? Well, if the types are convertible with each other (which is usually the case when you want to work with an heterogeneous container), a conversion will happen, and an homogeneous comparison will be performed instead. 

----
class MyString {
public:
  operator std::string() const {return myData;} // Converts to a string
  // Other functions to make the class behaves correctly
private:
  char *myData;
};

std::map<std::string, int, std::less<>> m;
MyString str{"abc"};
m.find(str);
----

In this case, ``++str++`` will not be converted to a ``++std::string++`` when calling the function ``++find++`` (this function is now a member function template). However, each time during the search when a comparison will be needed between ``++MyString++`` and ``++std::string++``, a conversion will now happen to convert ``++str++`` to always the same ``++string++``. A unique conversion is replaced by ``++O(ln N)++`` conversions.

This rule raises an issue when a transparent function of an heterogeneous container is used with a type that cannot be directly compared with the container key type.


== Noncompliant Code Example

----
void f() {
  std::map<std::string, int, std::less<>> m;
  MyString str{"abc"}; // See previous definition of MyString
  m.find(str); // Noncompliant, O(ln N) conversions
}
----


== Compliant Solution

Option 1: Make the container non heterogeneous

----
void f() {
  std::map<std::string, int> m;
  MyString str{"abc"}; // See previous definition of MyString
  m.find(str); // Compliant, one conversion at the start
}
----
Option 2: Provide heterogeneous comparisons

----

bool operator<(MyString const &s1, std::string const &s2) {/*...*/}
bool operator<(std::string const &s1, MyString const &s2) {/*...*/}
void f() {
  std::map<std::string, int, std::less<>> m;
  MyString str{"abc"}; // See previous definition of MyString
  m.find(str); // Compliant, no conversion at all
}
----


