include::../description.adoc[]

== Noncompliant Code Example

----
Try
    ' Some work which end up throwing an exception
    Throw New ArgumentException()
Finally
    ' Clean up
    Throw New InvalidOperationException()       ' Noncompliant; will mask the ArgumentException
End Try
----

== Compliant Solution

----
Try
    ' Some work which end up throwing an exception
    Throw New ArgumentException()
Finally
    ' Clean up
End Try
----

include::../see.adoc[]
