The ``++nodiscard++`` attribute can be used with or without explanations, but explaining why a result should not be discarded can only improve oneâ€™s understanding of the code, and would prevent developers from wasting time figuring those things out by themselves.


This rule raises an issue when ``++nodiscard++`` is used without any explanation.


== Noncompliant Code Example

----
enum class [[nodiscard]] ErrorCode { // Noncompliant
  OK,
  FAIL
};

ErrorCode computeData();
----


== Compliant Solution

----
enum class [[nodiscard("Check error code to know if the operation succeeded")]] ErrorCode {
  OK,
  FAIL
};

ErrorCode computeData();
----

