include::../description.adoc[]

== Noncompliant Code Example

----
do {
    try loadImage(imagePath)
} catch { // Noncompliant
    throw error
}
----

== Compliant Solution

----
do {
    try loadImage(imagePath)
} catch {
    handleImageError(error)
    throw error
}
----
or

----
try loadImage(imagePath)
----

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
