include::../description.adoc[]

== Noncompliant Code Example

----
var result = numerator / denominator  // FIXME denominator value might be  0
----

include::../see.adoc[]

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
