include::../description.adoc[]

== Noncompliant Code Example

----
if (True): print("hello")
----

== Compliant Solution

----
if (True): 
    print("hello")
----

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
