=== On 2014-09-19T13:39:33Z Freddy Mallet Wrote:
@Ann, I would improve the "Noncompliant Code Example" because when we don't know that's the type of time_limit variable is int, the example is meaningless. 

