include::../description.adoc[]

== Noncompliant Code Example

----
$name = $_GET["name"];
echo "Welcome $name"; // Noncompliant
----

== Compliant Solution

----
$name = $_GET["name"];
$safename = htmlspecialchars($name);
echo "Welcome $safename";
----

include::../see.adoc[]

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
