When a closure contains only a ``++return++`` statement, the ``++return++`` itself can be omitted.

== Noncompliant Code Example

----
someList.sort { a, b in
  return a > b
}
----

== Compliant Solution

----
someList.sort { a, b in a > b }
----
