include::../description.adoc[]

== Noncompliant Code Example

With the default regular expression ``++^[A-Z][a-zA-Z0-9]*$++``:

----
interface myInterface {...} // Noncompliant
----

== Compliant Solution

----
interface MyInterface {...}
----

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
