include::../description.adoc[]

== Noncompliant Code Example

With the default regular expression ``^[a-z][a-zA-Z0-9]*$``:

----
class MyClass {
   private int my_field;
}
----

== Compliant Solution

----
class MyClass {
   private int myField;
}
----
