include::../description.adoc[]

== Noncompliant Code Example

----
for (var i = 0; i < 10; i++) {
  ...
  i = i - 1; // Noncompliant 
  ...
} 

for (var i = 0; i < getMaximumNumber(); i++) {...} 
----

== Compliant Solution

----
int stopCondition = getMaximumNumber();
for (var i = 0; i < stopCondition; i++) {...} 
----

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
