include::../description.adoc[]

== Noncompliant Code Example

----
Public Class Foo

    Private name As String = "foobar" ' Noncompliant

    Public ReadOnly Property DefaultName As String = "foobar" ' Noncompliant

    Public Sub Foo(Optional Value As String = "foobar") ' Noncompliant

        Dim Something = If(value, "foobar") ' Noncompliant

    End Sub

End Class
----

== Compliant Solution

----
Public Class Foo

    Private Const Foobar As String = "foobar"

    Private name As String = Foobar

    Public ReadOnly Property DefaultName As String = Foobar

    Public Sub Foo(Optional Value As String = Foobar)

        Dim Something = If(value, Foobar);

    End Sub

End Class
----

== Exceptions

The following are ignored:

* literals with fewer than 5 characters
* literals matching one of the parameter names
* literals used in attributes
