According to the PSR2 coding standard:
____
The keyword ``elseif`` SHOULD be used instead of ``else if`` so that all control keywords look like single words.
____


== Noncompliant Code Example

----
if ($expr1) {
  ...
} else if ($expr2) {
  ...
} else {...}
----


== Compliant Solution

----
if ($expr1) {
  ...
} elseif ($expr2) {
  ...
} else {...}
----

