Python has no pre/post increment/decrement operator. For instance, ``++x++++`` and ``++x--++`` will fail to parse. More importantly, ``++++x++`` and ``++--x++`` will do nothing. To increment a number, simply write ``++x += 1++``.


== Noncompliant Code Example

----
++x # Noncompliant
----


== Compliant Solution

----
x += 1
----

