include::../description.adoc[]

== Noncompliant Code Example

----
func main() {
  v1 := (true && false) && (true && false) // Noncompliant
}
----

== Compliant Solution

----
func main() {
  v1 := (true && false) // Compliant
}
----

== Exceptions

This rule ignores ``++*++``, ``+++++``, ``++<<++`` and ``++=++``. 

include::../see.adoc[]
