=== On 2016-01-07T20:16:35Z Alban Auzeill Wrote:
In the bellow description there's nothing about {color:#8eb021}protected{color} inheritance. What I've done in rule's implementation, is applying the same logique to {color:#8eb021}private{color} and {color:#8eb021}protected{color}.

Do we have to change the documentation about this?


I also distinguish the default inheritance of a class as private and  the default inheritance of a struct as public.

My non-compliant test cases are:

----
class B : A {
};

class C : protected A {
};

struct D : protected A {
};

class E : private A {
};

struct F : private A {
};
----

My compliant test cases are:

----
struct B : A {
};

class C : public A {
};

struct D : public A {
};
----

Do you agree?

=== On 2016-01-07T20:40:38Z Ann Campbell Wrote:
Since both the title and description are about "non-public" inheritance, I think the ``++protected++`` case is already adequately covered. Note that the Noncompliant Code Example only shows ``++private++`` inheritance because adding a ``++protected++`` example wouldn't have contributed anything significant.

