include::../description.adoc[]

== NonCompliant Code Example

----
function process(var a) {
  var BAD_OPS = 3;
  if( a = BAD_OPS) {...}
}
----


----
function process(var a) {
  var BAD_OPS = "UNKNOWN";
  if( a = BAD_OPS) {...}
}
----

include::../see.adoc[]
