include::../description.adoc[]

== Noncompliant Code Example

----
<!-- TODO Drop this dependency -->
<dependency>
  <groupId>org.apache.commons</groupId>
  <artifactId>commons-lang3</artifactId>
  <version>3.8.1</version>
</dependency>
----

include::../see.adoc[]
