include::../description.adoc[]

== Noncompliant Code Example

----
if (condition)  // Noncompliant
  executeSomething()
----

== Compliant Solution

----
if (condition) {
  executeSomething()
}
----

include::../see.adoc[]

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
