This rule raised an issue when a component is used in the ``++render()++`` method and not ``++import++``ed or defined. 


== Noncompliant Code Example

----
<Rating value={this.props.measure.reliability_rating} />;
----


== Compliant Solution

----
import Rating from './Rating'
[...]
<Rating value={this.props.measure.reliability_rating} />
----

