According to the documentation, AssertJ https://tinyurl.com/yxnzt6pj[isInstanceOfSatisfying]...

??verifies that the actual value is an instance of the given type satisfying the given requirements expressed as a Consumer.??

The second argument is a `+Consumer+`, not a predicate, it does not check the return value of the function, it assumes the function will do assertions itself. If you do not do any assertion in the `+Consumer+`, it probably means that you are inadvertently only partially testing your object.

This rule raises an issue when the second argument of `+isInstanceOfSatisfying+` does not contain any assertion.


== Noncompliant Code Example

----
assertThat(myObject).isInstanceOfSatisfying(String.class, s -> "Hello".equals(s)); // Not testing the String value.
----


== Compliant Solution

----
assertThat(myObject).isInstanceOfSatisfying(String.class, s -> assertThat(s).isEqualTo("Hello"));
----

