include::../description.adoc[]

== Noncompliant Code Example

With the default regular expression ``++@[a-zA-Z0-9_]*++``:

----
CREATE PROCEDURE proc1
    @@var1 INT -- Noncompliant
AS
BEGIN
    DECLARE @@var2 INT; -- Noncompliant
END
----
