include::../description.adoc[]

== Noncompliant Code Example

----
if (booleanMethod() || false) { /* ... */ }
doSomething(!false);
doSomething(booleanMethod() && true);
----

== Compliant Solution

----
if (booleanMethod()) { /* ... */ }
doSomething(true);
doSomething(booleanMethod());
----
