include::../description.adoc[]

== Noncompliant Code Example

----
if (booleanMethod() || false) { /* ... */ }
doSomething(!false);
doSomething(booleanMethod() && true);
----

== Compliant Solution

----
if (booleanMethod()) { /* ... */ }
doSomething(true);
doSomething(booleanMethod());
----

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
