This rule allows you to track the use of the PMD suppression comment mechanism. 


== Noncompliant Code Example

----
// NOPMD
----


ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::rspecator-view[]
