include::../description.adoc[]

== Noncompliant Code Example

----
try:
  result = do_the_thing(a)
except:  #Noncompliant; no specific exception caught
  result = None

try:
  result = do_the_thing(a)
except Exception:  #Noncompliant; too generic to be the only exception type caught
  result = None
----

== Compliant Solution

----
try:
  result = do_the_thing(a)
except ZeroDivisionError:
  result = None
except TypeError:
  result = -1
----

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
