include::../description.adoc[]

== Noncompliant Code Example

----
// The following string will error if there is a whitespace after '\' 
var str = "Hello \ 
World"; 
----

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
