include::../description.adoc[]

== Noncompliant Code Example

----
// Noncompliant; there are extra spaces after 'String'
var str:String     
----

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
