Arrays in JavaScript have several methods for filtering, mapping or folding that require a callback. Not having a return statement in such a callback function is most likely a mistake.

This rule applies for the following methods of an array:

* ``++Array.from++``
* ``++Array.prototype.every++``
* ``++Array.prototype.filter++``
* ``++Array.prototype.find++``
* ``++Array.prototype.findIndex++``
* ``++Array.prototype.map++``
* ``++Array.prototype.reduce++``
* ``++Array.prototype.reduceRight++``
* ``++Array.prototype.some++``
* ``++Array.prototype.sort++``

== Noncompliant Code Example

----
var merged = arr.reduce(function(a, b) {
  a.concat(b);
}); // Noncompliant: No return statement
----

== Compliant Solution

----
var merged = arr.reduce(function(a, b) {
  return a.concat(b);
});
----
