include::../description.adoc[]

== Noncompliant Code Example

----
if (x == 0) 
{
    DoSomething();
} 
else if (x == 1) 
{
    DoSomethingElse();
} 
----

== Compliant Solution

----
if (x == 0) 
{
    DoSomething();
} 
else if (x == 1) 
{
    DoSomethingElse();
} 
else 
{
    throw new InvalidOperationException();
}
----

== Exceptions

None

include::../see.adoc[]

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
