Multiplying ``++sizeof()++`` with ``++sizeof()++`` indicates a logic error.


== Noncompliant Code Example

----
sizeof(a) * sizeof(b); // Noncompliant
----


ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::rspecator-view[]
