include::../description.adoc[]

== Noncompliant Code Example

With the default regular expression `+^[a-z][a-zA-Z0-9]*$+`:

----
class MyClass {
  int my_field;
};
----

== Compliant Solution

----
class MyClass {
  int myField;
};
----
