include::../description.adoc[]

== Noncompliant Code Examples

For https://github.com/GehirnInc/python-jwt[jwt] module:

----
jwt.decode(token, verify = False)  # Noncompliant
----

----
try:
    jwt.decode(token, key, algo)  # Noncompliant; generic exception catch
except:
    pass
----

For https://github.com/davedoesdev/python-jwt[python_jwt] module:

----
jwt.process_jwt(token)  # Noncompliant
----

== Compliant Solution

For https://github.com/GehirnInc/python-jwt[jwt] module:

----
jwt.decode(token, key, algo)  # Compliant
----

----
try:
    jwt.decode(token, key, algo)
except jwt.InvalidSignatureError:  # Compliant
    pass
----

For https://github.com/davedoesdev/python-jwt[python_jwt] module:

----
jwt.verify_jwt(token, key, algo)  # Compliant
----

include::../see.adoc[]
