It can be extremely confusing when a `+for+` condition tests a variable which is not updated inside the `+for+` post statement.

== Noncompliant Code Example

----
for i := 1; i <= 10; j++ { // Noncompliant
	// ...
}
----

== Compliant Solution

----
for i := 1; i <= 10; i++ {
	// ...
}
----
