The ``asm`` declaration is available to all {cpp} implementations, allowing a consistent mechanism to be used.

However, the parameters to ``asm`` are still implementation-defined.


== Noncompliant Code Example

----
void Delay ( void )
{
#pragma asm
  "NOP" // Noncompliant
#pragma endasm
}
----


== Compliant Solution

----
void Delay ( void )
{
  asm ( "NOP" ); // Compliant
}
----


== See

* MISRA {cpp} 2008, 7-4-2

