include::../description.adoc[]

== Noncompliant Code Example

----
if someCondition; puts "hello"; end
----

== Compliant Solution

----
if someCondition
  puts "hello"
end
----

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
