include::../description.adoc[]

== Noncompliant Code Example

----
public void doSomething() {                
  super.doSomething();
}

@Override
public boolean isLegal(Action action) {      
  return super.isLegal(action);
}
----

== Compliant Solution

----
@Override
public boolean isLegal(Action action) {         // Compliant - not simply forwarding the call
  return super.isLegal(new Action(/* ... */));
}

@Id
@Override
public int getId() {                            // Compliant - there is annotation different from @Override
  return super.getId();
}
----

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
