include::../description.adoc[]

== Noncompliant Code Example

----
public string GetReadableStatus(Job j)
{
  return j.IsRunning ? "Running" : j.HasErrors ? "Failed" : "Succeeded";  // Noncompliant
}
----

== Compliant Solution

----
public string GetReadableStatus(Job j) 
{
  if (j.IsRunning) 
  {
    return "Running";
  }
  return j.HasErrors ? "Failed" : "Succeeded";
}
----
