// :language: cobol
// :file_extension: cbl
// :language_group: cobol
// :cobol:

// :language: java
// :file_extension: java
// :language_group: java

// :language: php
// :language_group: php
// :file_extension: php

:source-highlighter: prettify
:source-language: {language}
:source-indent: 0

Duplicated string literals make the process of refactoring error-prone, since you must be sure to update all occurrences.
On the other hand, constants can be referenced from many places, but only need to be updated in a single place.

== Noncompliant Code Example
With the default threshold of 3:
----
include::{language_group}/noncompliant.{file_extension}[tag=include]
----

== Compliant Code Example
----
include::{language_group}/compliant.{file_extension}[tag=include]
----

== Exceptions
ifdef::cobol[]
include::{language_group}/exceptions.adoc[]
endif::[]

ifndef::cobol[]
To prevent generating some false-positives, literals having less than 5 characters are excluded.
endif::[]