include::../description.adoc[]

== Noncompliant Code Example

----
IF something
  IF something_else
    -- ...
----

== Compliant Solution

----
IF something AND something_else
  -- ...
----

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
