include::../description.adoc[]

== Noncompliant Code Example

----
if (GetTrue() | GetFalse()) // Noncompliant; both sides evaluated
{
}
----

== Compliant Solution

----
if (GetTrue() || GetFalse()) // true short-circuit logic
{
}
----

include::../see.adoc[]

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
