include::../description.adoc[]

== Noncompliant Code Example

----
Imports System.Reflection

<Assembly: AssemblyTitle("MyAssembly")> ' Noncompliant

Namespace MyLibrary

End Namespace
----

== Compliant Solution

----
Imports System.Reflection

<Assembly: AssemblyTitle("MyAssembly")>
<Assembly: AssemblyVersion("1.1.125.0")>

Namespace MyLibrary

End Namespace
----

include::../see.adoc[]
