include::../description.adoc[]

== Noncompliant Code Example

With default provided regular expression ``^[A-Z][a-zA-Z0-9]*$``:

----
class my_class
  ...
end
----

== Compliant Solution

----
class MyClass
  ...
end
----
