include::../description.adoc[]

== Noncompliant Code Example

----
sampleString  = "Hello World" : MsgBox(sampleString)
----

== Compliant Solution

----
sampleString  = "Hello World"
MsgBox(sampleString)
----

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
