include::../description.adoc[]

== Noncompliant Code Example

----
function divide(numerator, denominator) {
  return numerator / denominator;              // FIXME denominator value might be  0
}
----

include::../see.adoc[]

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
