include::../description.adoc[]

== Noncompliant Code Example

----
# TODO 
----

== Compliant Solution

----
# TODO(ganncamp) per the business partners, more checks needed
----

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
