include::../description.adoc[]

== Noncompliant Code Example

----
if  !(a == 2)  # Noncompliant
  # ...
end

b = !(a < 10) # Noncompliant
----

== Compliant Solution

----
if a != 2
  # ...
end
b = (a >= 10)
----

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
