include::../description.adoc[]

== Noncompliant Code Example

----
fun doSomething() {
  // TODO
}
----

include::../see.adoc[]

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
