include::../description.adoc[]

== Noncompliant Code Example

With the default regular expression ``++^[a-z][a-zA-Z0-9]*$++``:

----
let Pi = 3.14
----

== Compliant Solution

----
let pi = 3.14
----
