Jump statements (``++return++``, ``++break++`` and ``++next++``) move control flow out of the current code block. So any statements that come after a jump are dead code. 

== Noncompliant Code Example

----
def foo(a)
  i = 10
  return a + i    # Noncompliant 
  i += 1          # dead code
end
----

== Compliant Solution

----
def foo(a)
  i = 10
  return a + i
end
----

include::../see.adoc[]
