include::../description.adoc[]

== Noncompliant Code Example

----
if (x != undefined) {
  if (y === 2) {
    // ...
  }
}
----

== Compliant Solution

----
if (x != undefined && y === 2) {
  // ...
}
----

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
