The ``void`` operator evaluates its argument and unconditionally returns ``undefined``. It can be useful in pre-ECMAScript 5 environments, where ``undefined`` could be reassigned, but generally, its use makes code harder to understand.


== Noncompliant Code Example

----
void (function() {
   ...
}());
----


== Compliant Solution

----
(function() {
   ...
}());
----


== Exceptions

No issue is raised when ``void 0`` is used in place of ``undefined``. 

----
if (parameter === void 0) {...}
----

