include::../description.adoc[]

== Noncompliant Code Example

----
towns[i] = "London";
towns[i] = "Chicago";  // Noncompliant
----

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
