The {cpp} standard defines some identifiers as having special meaning in some contexts. These are ``audit``, ``axiom``, ``final`` and ``override``. While it is technically possible to use them as normal identifiers, it's clearer for the reader of the code to consider them as if they were keywords, and only use them with their special meaning.

== Noncompliant Code Example

----
void axiom(int final); // Noncompliant
----

== Compliant Solution

----
void precept(int finalValue); // Compliant
----
