Use of the bang (`+!+`) operator leads to late binding and results in inefficient code. Use the slightly more verbose dot (`+.+`) notation instead.


== Noncompliant Code Example

----
MsgBox ObjectWithDefaultProperty!Param
----


== Compliant Solution

----
MsgBox ObjectWithDefaultProperty.DefaultProperty("Param")
----

