For better readability, do not put multiple property declarations in the same statement.


== Noncompliant Code Example

----
<?php
class Foo
{
   private $bar = 1, $bar2 = 2;
}
----


== Compliant Solution

----
<?php
class Foo
{
   private $bar1 = 1; 
   private $bar2 = 2;
}
----


ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::rspecator-view[]
