include::../description.adoc[]

== Noncompliant Code Example

----
def set_name(name)
  name = name
end
----

== Compliant Solution

----
def set_name(name)
  @name = name
end
----

include::../see.adoc[]

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
