=== Duplicate: RSPEC-827

=== Relates: RSPEC-806

=== Relates: RSPEC-830

=== Relates: RSPEC-832

=== On 2014-10-16T12:53:05Z Ann Campbell Wrote:
\[~samuel.mercier] please:

* fill in the appropriate reference field(s).
* provide a See section.
* use the standard section titles: Noncompliant Code Example, Exception*s*, and heading levels (h2. instead of h3.).
* use the standard section order: description, Noncompliant Code Example, Compliant Solution, Exceptions, See* SQALE.
* use our standard: "Noncompliant", rather than "Non-compliant" (code sample).


=== On 2014-10-17T15:51:08Z Ann Campbell Wrote:
\[~freddy.mallet] this an RSPEC-832 overlap to my mind.

=== On 2014-10-17T19:08:06Z Ann Campbell Wrote:
FYI [~samuel.mercier] I've moved this to Data-related reliability - the danger appears to be that a non-static var could be visible (and thus accidentally used!) outside its intended scope. Change it back if I'm wrong.

