In Spring applications, application components that expose interfaces should be package protected at most, not ``++public++``. Such reduced visibility helps ensure that the interface is only accessed through the container and not directly. 

== Noncompliant Code Example

----
// TODO
----

== Compliant Solution

----
// TODO
----
