include::../description.adoc[]

== Noncompliant Code Example

----
for i in range(3):
    pass
----

include::../exceptions.adoc[]

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
