include::../description.adoc[]

== Noncompliant Code Example

----
<html>                                    <!-- Noncompliant -->
...
</html>
----

== Compliant Solution

----
<!-- Copyright 2013 SonarSource SA -->
<html>
...
</html>
----

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
