include::../description.adoc[]

== Noncompliant Code Example

With default provided regular expression^[A-Z][a-zA-Z]*$:


----
AWSTemplateFormatVersion: 2010-09-09
Resources:
  S3Bucket:
    Type: 'AWS::S3::Bucket' 
    Properties: 
      BucketName: "mybucketname"
      Tags: 
        - Key: "anycompany:cost-center" # Noncompliant
          Value: "Accounting"
        - Key: "anycompany:EnvironmentType"
          Value: "PROD"
----
 

== Compliant Solution

 

----
AWSTemplateFormatVersion: 2010-09-09
Resources:
  S3Bucket:
    Type: 'AWS::S3::Bucket'
    Properties: 
      BucketName: "mybucketname"
      Tags: 
        - Key: "anycompany:CostCenter"
          Value: "Accounting"
        - Key: "anycompany:EnvironmentType"
          Value: "PROD"
----

include::../see.adoc[]
