include::../description.adoc[]

== Noncompliant Code Example

Given parameters:

* className:System.Console

----
System.Console.WriteLine("foo");  // Noncompliant
----

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
