include::../description.adoc[]

== Noncompliant Code Example

With default provided regular expression: ``++^(_|[a-zA-Z0-9]+)$++``:

----
func execute_all() {
...
}
----

== Compliant Solution

----
func executeAll() {
...
}
----

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
