include::../description.adoc[]

== Noncompliant Code Example

With default provided regular expression: ``^[A-Z][a-zA-Z0-9]*``

----
enum someEnumeration { // Non-Compliant
    case Bar
}
----

== Compliant Solution

----
enum SomeEnumeration {
    case Bar
}
----
