== Noncompliant Code Example

Using the default importing  and returning regular expressions ``I_\[a-zA-Z_&#92;-0-9\]+``, and ``R_\[a-zA-Z_&#92;-0-9\]+``:

----
    METHODS m_fact IMPORTING i1  TYPE i
                       value(i2) TYPE i
                   RETURNING value(factorial) TYPE i.
----
