include::../description.adoc[]

== Noncompliant Code Example

----
if (x == 0) {
  doSomething();
} else if (x == 1) {
  doSomethingElse();
}
----

== Compliant Solution

----
if (x == 0) {
  doSomething();
} else if (x == 1) {
  doSomethingElse();
} else {
  throw "Unexpected value for x";
}
----

include::../see.adoc[]

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
