According to the specification:
bq. Nonfinal ``++stati{cpp}`` class fields are disallowed in EJBs because such fields make an enterprise bean difficult or impossible to distribute.

Therefore, all ``++stati{cpp}`` fields in an EJB should also be ``++final++``.


== Noncompliant Code Example

----
@Stateless
public class MyEjb
{
  private static String message;  // Noncompliant
}
----

