According to the specification:

bq. Nonfinal ``++static++`` class fields are disallowed in EJBs because such fields make an enterprise bean difficult or impossible to distribute.


Therefore, all ``++static++`` fields in an EJB should also be ``++final++``.


== Noncompliant Code Example

----
@Stateless
public class MyEjb
{
  private static String message;  // Noncompliant
}
----

