include::../description.adoc[]

== Noncompliant Code Example

----
Namespace MyLibrary  ' Noncompliant

End Namespace
----

== Compliant Solution

----
<Assembly: Runtime.InteropServices.ComVisible(False)>

Namespace MyLibrary

End Namespace
----
