As stated in the Glibc documentation:

____
The behavior of the 's[n]printf()' functions are undefined if copying takes place between objects that overlap—for example, if s is also given as an argument to be printed under control of the ‘%s’ conversion

____


== Noncompliant Code Example

----
sprintf(buf, "%s Entry: %d\n", buf, i); //Noncompliant 
----

