include::../description.adoc[]

== Noncompliant Code Example

----
if (a) {
  if (a) { // Noncompliant
    doSomething();
  }
}
----

== Compliant Solution

----
if (a) {
  if (b) {
    doSomething();
  }
}

// or 
if (a) {
  doSomething();
}
----

include::../see.adoc[]

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
