include::../description.adoc[]

== Noncompliant Code Example

----
using (var tripleDES = new TripleDESCryptoServiceProvider()) //Noncompliant
{
  //...
}
----

== Compliant Solution

----
using (var aes = new AesCryptoServiceProvider())
{
  //...
}
----

include::../see.adoc[]

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
