include::../description.adoc[]

== Noncompliant Code Example

----
for ($i = 0; $i < $length; $i--) { // Noncompliant
  //...
}
----

== Compliant Solution

----
for ($i = 0; $i < $length; $i++) {
  //...
}
----

include::../see.adoc[]

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
