This statement deletes all rows of an internal table itab. This ``++REFRESH++`` statement is deprecated and usage should be avoided.


== Noncompliant Code Example

----
REFRESH itab.
----


== Compliant Solution

----
CLEAR itab.
----

