include::../description.adoc[]

== Noncompliant Code Example

----
if boolFunc() || false { 
    // ...
}

flag := x && true
----

== Compliant Solution

----
if boolFunc() {
    // ...
}

flag := x
----
