include::../description.adoc[]

== Noncompliant Code Example

----
public void SetName(string name)
{
  name = name;
}
----

== Compliant Solution

----
public void SetName(string name)
{
  this.name = name;
}
----

include::../see.adoc[]

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
