include::../description.adoc[]

== Noncompliant Code Example

----
CREATE PROCEDURE doSomething
	AS
BEGIN
  ...
  -- TODO something
  ...
END
GO
----

include::../see.adoc[]

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
