include::../description.adoc[]

== Noncompliant Code Example

With the default regular expression ``++YIF_[A-Za-z]+++``:

----
INTERFACE MyInterface. "Noncompliant
...
ENDINTERFACE.
----

== Compliant Solution

----
INTERFACE YIF_myinterface. 
...
ENDINTERFACE.
----

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
