include::../description.adoc[]

== Noncompliant Code Example

With the default regular expression ``YIF_[A-Za-z]+``:

----
INTERFACE MyInterface. "Noncompliant
...
ENDINTERFACE.
----

== Compliant Solution

----
INTERFACE YIF_myinterface. 
...
ENDINTERFACE.
----
