include::../description.adoc[]

== Noncompliant Code Example

With default provided regular expression ``++^[A-Z][a-zA-Z0-9_]*$++``:

----
class myClass {}
----

== Compliant Solution

----
class MyClass {}
----
