Using ``++[Int]().count++`` to test for emptiness works, but using ``++[Int]().isEmpty++`` makes the code more readable and can be more performant. The time complexity of any ``++isEmpty++`` implementation should be ``++O(1)++`` whereas some implementations of ``++count()++`` can be ``++O(n)++``.

== Noncompliant Code Example

----
if (arr.count == 0) {  // Noncompliant
  /* ... */
}
----

== Compliant Solution

----
if (arr.isEmpty) {
  /* ... */
}
----
