include::../description.adoc[]

== Noncompliant Code Example

With the default regular expression ``++^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$++``:

----
define("const1", true);

class Foo {
    const const2 = "bar";
}
----

== Compliant Solution

----
define("CONST1", true);

class Foo {
    const CONST2 = "bar";
}
----
