=== On 2014-12-15T13:46:44Z Ann Campbell Wrote:
assigned to you for approval

=== On 2014-12-15T14:17:18Z Pierre-Yves Nicolas Wrote:
Looks good to me.

I guess that rule should be fired for a block of lines of code rather than for individual lines. If that's the case, then we may have to use a linear remediation function.

=== On 2014-12-15T14:49:53Z Ann Campbell Wrote:
Linear with offset based on the number of lines? What do you think [~pierre-yves.nicolas]? 

5 min + 5min/line ?


For the noncompliant code example, that works out to 20min...

=== On 2014-12-15T16:59:31Z Pierre-Yves Nicolas Wrote:
Ok. We should probably update the message to make it clear that the issue is on a block.

