include::../description.adoc[]

== Noncompliant Code Example

----
try {
  doSomething();
} catch (ex) {  // Noncompliant
  throw ex;
}
----

== Compliant Solution

----
try {
  doSomething();
} catch (ex) {
  console.err(ex);
  throw ex;
}
----
or

----
doSomething();
----

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
