include::../description.adoc[]

== Noncompliant Code Example

----
foo(); bar() // Noncompliant
----

== Compliant Solution

----
foo()
bar()
----

== Exceptions

Control flow statements with a single nested statement are ignored.


----
if condition { doSomething() } // Compliant
----

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
