include::../description.adoc[]

== Noncompliant Code Example

----
Connection conn = DriverManager.getConnection("jdbc:derby:memory:myDB;create=true", "login", "");
----

== Compliant Solution

----
String password = System.getProperty("database.password");
Connection conn = DriverManager.getConnection("jdbc:derby:memory:myDB;create=true", "login", password);
----

include::../see.adoc[]
