include::../description.adoc[]

== Noncompliant Code Example

----
towns.Item(x) = "London"
towns.Item(x) = "Chicago";  // Noncompliant
----

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
