Backticks are a deprecated alias for ``++repr()++``. Don't use them any more, the syntax was removed in Python 3.0.


== Noncompliant Code Example

----
return `num`  # Noncompliant
----


== Compliant Solution

----
return repr(num)
----

