include::../description.adoc[]

== Noncompliant Code Example

----
class MyClass {
public :
  static void Mymethod() {
    // ...
  }
};

MyClass* pmyclass = new MyClass();
pmyclass->Mymethod(); // Noncompliant
----

== Compliant Solution

----
class MyClass {
public :
  static Mymethod() {
    // ...
  }
};

Myclass::Mymethod();
----

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
