include::../description.adoc[]

== Noncompliant Code Example

With the default regular expression ``++^[a-z][a-zA-Z0-9]*$++``:

----
void doSomething(int my_param) {
  int LOCAL;
  ...
}
----

== Compliant Solution

----
void doSomething(int myParam) {
  int local;
  ...
}
----

== Exceptions

This rule ignores ``++const++`` variables.

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
