
== Noncompliant Code Example

----
var i, j int = 1, 2
----

== Compliant Solution

----
var i int = 1
var j int = 2
----

include::../see.adoc[]

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
