include::../description.adoc[]

== Noncompliant Code Example

----
Select Case variable
    Case 0
        doSomething()
    Case Else
        doSomethingElse()
End Select
----

== Compliant Solution

----
If variable = 0 Then
    doSomething()
Else
    doSomethingElse()
End If
----
