include::../description.adoc[]

== Noncompliant Code Example

----
CREATE TABLE mytable(col1 INTEGER NOT NULL);
INSERT INTO mytable VALUES (value) -- FIXME ensure value is not null
----

include::../see.adoc[]

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
