The W3C specifications define the valid `+at-rules+`. Only the official and browser-specific `+at-rules+` should be used to get the expected impact in the final rendering.


== Noncompliant Code Example

----
@encoding "utf-8";
----


== Compliant Solution

----
@charset "utf-8";
----

