include::../description.adoc[]

== Noncompliant Code Example

With default provided regular expression ^[A-Z][a-zA-Z]*$:


----
resource "aws_s3_bucket" "mynoncompliantbucket" {
  bucket = "mybucketname"

  tags = {
    anycompany:cost-center = "Accounting" # Noncompliant
  }
}
----

== Compliant Solution

----
resource "aws_s3_bucket" "mycompliantbucket" {
  bucket = "mybucketname"

  tags = {
    anycompany:CostCenter = "Accounting"
  }
}
----

include::../see.adoc[]
