=== Relates: RSPEC-1155

=== On 2014-12-02T07:35:56Z Nicolas Peru Wrote:
Doesn't this one duplicates this : \http://jira.sonarsource.com/browse/RSPEC-1155 ?

=== On 2014-12-02T14:44:21Z Ann Campbell Wrote:
It does now [~nicolas.peru]. It was originally written as "just don't use ``++ConcurrentLinkedQueue++`` - for anything!" in response to an internal email thread.

\[~freddy.mallet]...?

=== On 2014-12-04T10:15:47Z Freddy Mallet Wrote:
Ok [~nicolas.peru] and [~ann.campbell.2], I've changed the content of the noncompliant code example to prevent any misunderstanding but for me there is no overlap between RSPEC-2250 and RSPEC-1155.

