This rule allows banning some modules.

== Noncompliant Code Example

With ``++moduleName++`` configured with ``++[a-zA-Z-]*UT123[a-zA-Z]*++``:


----
CALL UT123.
CALL UT123L.
CALL UT123-L.
CALL WS-UT123.
----
