include::../description.adoc[]

== Noncompliant Code Example

With the default regular expression ``++^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$++``:

----
Global Const WdgtStatusOk = 0
----

== Compliant Solution

----
Global Const WDGT_STATUS_OK = 0
----
