include::../description.adoc[]

== Noncompliant Code Example

----
Public Function GetReadableStatus(job As Job) As String
    Return If(job.IsRunning, "Running", If(job.HasErrors, "Failed", "Succeeded")) ' Noncompliant
End Function
----

== Compliant Solution

----
Public Function GetReadableStatus(job As Job) As String
    If job.IsRunning Then Return "Running"
    Return If(job.HasErrors, "Failed", "Succeeded")
End Function
----
