include::../description.adoc[]

== Noncompliant Code Example

With a maximum number of 4 parameters:

----
func doSomething(param1: String, param2: String, param3: String, param4: String, param5: String) {
    // ...
}
----

== Compliant Solution

----
func doSomething(param1: String, param2: String, param3: String, param4: String) {
    // ...
}
----

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::rspecator-view[]
