This rule checks that the specified child tag does not appear as a direct child of the specified parent.


== Noncompliant Code Example

Assuming a parent/child combination of ``head``/``body``:

----
<head>
  ...
  <body>  <!-- Noncompliant -->
    ...
  </body>
</head>
----


== Compliant Solution

----
<head>
  ...
</head>
----

