include::../description.adoc[]

== Noncompliant Code Example

With the default regular expression ``++^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$++``:

----
public static const first:String = "first";
----

== Compliant Solution

----
public static const FIRST:String = "first";
----
