include::../description.adoc[]

== Noncompliant Code Example

----
import CryptoSwift

let bytes:Array<UInt8> = [0x01, 0x02, 0x03]
let digest = input.md5() // Noncompliant
----

== Compliant Solution

----
import CryptoSwift

let bytes:Array<UInt8> = [0x01, 0x02, 0x03]
let digest = input.sha256() // Compliant
----

include::../see.adoc[]
