include::../description.adoc[]

== Noncompliant Code Example

With the default provided regular expression: ``++^[a-z_][a-z0-9_]*$++``

----
def MyFunction(a,b):
    ...
----

== Compliant Solution

----
def my_function(a,b):
    ...
----

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
