``GOTO`` is an unstructured control flow statement. It makes code less readable and maintainable, and should only be used to branch to defined ``ENDSR`` points within the source. For normal loop functions use ``ITER`` or ``LEAVE`` instead.

== Noncompliant Code Example

----
C     LOOP1         TAG 
... 
C                   GOTO      LOOP1 
----
