Using the standard `+getClassLoader()+` may not return the _right_ class loader in a JEE context. Instead, go through the `+currentThread+`.


== Noncompliant Code Example

----
ClassLoader cl = this.getClass().getClassLoader();  // Noncompliant
----


== Compliant Solution

----
ClassLoader cl = Thread.currentThread().getContextClassLoader();
----

