include::../description.adoc[]

== Noncompliant Code Example

----
int <T> Add(int a, int b) // Noncompliant; <T> is ignored
{
  return a + b; 
}
----

== Compliant Solution

----
int Add(int a, int b)
{
  return a + b; 
}
----

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
