A serialVersionUID field is required in a ``++Serializable++`` class. In a non-``++Serializable++``, it's just confusing.


== Noncompliant Code Example

----
public class MyClass {
  private static final long serialVersionUID = -1L;  // Noncompliant
}
----


ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::rspecator-view[]
