include::../description.adoc[]

== Noncompliant Code Example

----
<html>
<!-- TODO support small device -->
...
</html>
----

include::../see.adoc[]

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
