include::../description.adoc[]

== Noncompliant Code Example

----
public class Greeter {
  public static var foo:Foo = new Foo(...);
  ...
}
----

== Compliant Solution

----
public class Greeter {
  public static const FOO:Foo = new Foo(...);
  ...
}
----

include::../see.adoc[]

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
