The storage of a ``++FIXED DECIMAL++`` is most efficient when you use an odd number of digits, so that the leftmost byte is fully used.

== Noncompliant Code Example

----
 declare total fixed dec(10);
----

== Compliant Solution

----
 declare total fixed dec(9);
----
