include::../description.adoc[]

== Noncompliant Code Example

----
DATA: MyText TYPE string.
----

== Compliant Solution

----
DATA: my_text TYPE string.
----

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
