include::../description.adoc[]

== Noncompliant Code Example

----
for (var i = 0; i < length; i++) {}  // Empty on purpose or missing piece of code ?
----

== Exceptions

When a block contains a comment, this block is not considered to be empty. Moreover `+catch+` blocks are ignored.
