include::../description.adoc[]

== Noncompliant Code Example

----
try 
{
  /* some work which end up throwing an exception */
  throw new ArgumentException();
} 
finally 
{
  /* clean up */
  throw new InvalidOperationException();       // Noncompliant; will mask the ArgumentException
}
----

== Compliant Solution

----
try
{
  /* some work which end up throwing an exception */
  throw new ArgumentException();
} 
finally 
{
  /* clean up */                       // Compliant
}
----

include::../see.adoc[]

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
