``Path.GetTempFileName()`` generates predictable file names and is inherently unreliable and insecure. Additionally, the method will raise an ``IOException`` if it is used to create more than 65535 files without deleting previous temporary files.

Instead, ``Path.GetRandomFileName()`` should be used.

== Noncompliant Code Example

----
var tempPath = Path.GetTempFileName();
using (var writer = new StreamWriter(tempPath)) // Noncompliant
{
    writer.WriteLine("content");
}
----

== Compliant Solution

----
var randomPath = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
using (var writer = new StreamWriter(randomPath)) // Compliant
{
    writer.WriteLine("content");
}
----

include::../see.adoc[]
