Both `+php_sapi_name()+` and the `+PHP_SAPI+` constant give the same value. But calling the method is less efficient that simply referencing the constant. 


== Noncompliant Code Example

----
if (php_sapi_name() == 'test') { ... } 
----


== Compliant Solution

----
if (PHP_SAPI == 'test') { ... } 
----

