=== On 2015-03-25T16:03:23Z Tamas Vajk Wrote:
\[~ann.campbell.2] should I add the See section for C# as well?

=== On 2015-03-25T16:24:48Z Ann Campbell Wrote:
\[~tamas.vajk] these language-specific subtasks override portions of their parent tasks, and the description field is treated as 5 separate parts: description, Noncompliant Code Example, Compliant Solution, Exceptions, and references (See). There's no need to copy/paste the references to this child ticket, but do include them in the implementation's rule description.

include::../comments-and-links.adoc[]
