{cpp}20 introduces Concepts, a dedicated syntax for constraints on template parameters. Concepts enable you to express readable and modular constraints on template functions and template classes and their member functions. They replace the SFINAE-based approaches such as ``++std::enable_if++``.


When compared to SFINAE-based constraints Concepts have many advantages:

* Concepts allow constraining function arguments without turning off deduction and without disturbing arity
* Concepts make it much easier to write overloads with mutually exclusive constraints
* Concepts enable a broad range of constraint designs
* Concepts are easier to write, read, and allowed the compiler to generate better error messages

This rule reports the uses of ``++std::enable_if++`` that can be replaced by a "Concept" constraint.

== Noncompliant Code Example

----
template <typename N, class = typename std::enable_if<std::is_arithmetic_v<N>>::type> // Noncompliant
auto negate(N n) { return -n; }
----

== Compliant Solution

----
template <typename N>
  requires std::is_arithmetic_v<N>
auto negate(N n) { return -n; }
----

== See

http://open-std.org/JTC1/SC22/WG21/docs/papers/2016/p0225r0.html[Why I want Concepts, and why I want them sooner rather than later]
