include::../description.adoc[]

== Noncompliant Code Example

----
UPDATE person
SET name = name;
----

== Compliant Solution

----
UPDATE person
SET name = UPPER(name);
----

include::../see.adoc[]

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
