The result of an expression with an arithmetic operator ``/, *, %, {plus}{plus}, --, -, {plus}=, -=, *=, /=, %=, {plus}`` or unary operator ``+++, -++`` when at least one operand is ``++Object++`` or ``++Undefined++`` will be always a ``++NaN++`` (Not a Number).


== Noncompliant Code Example

----
x = [1, 2];
var y = x / 4;  //Noncompliant
----


== Exceptions

* ``++Date++`` operands: they are implicitly converted to numbers.
* The binary ``+`` operator with ``++Object++`` operand (concatenation).

