HTML-style comments are not part of EcmaScript specification, and should not be used.


== Noncompliant Code Example

----
<!-- Noncompliant -->
----


== Compliant Solution

----
// Compliant
/* Compliant */
----


ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::rspecator-view[]
