Having characters before ``++<?php++`` can cause "Cannot modify header information" errors and similar problems with Ajax requests.


== Noncompliant Code Example

----
test<?php  //Noncompliant
// ...
----
and

----
// Noncompliant; newline before opening tag
<?php
// ...
----


== Compliant Solution

----
<?php  
// ...
----



ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::rspecator-view[]
