include::../description.adoc[]

== Noncompliant Code Example

----
void doSomething(Integer a, Integer b) {     // "b" is unused
  compute(a);
}
----

== Compliant Solution

----
void doSomething(Integer a) {
  compute(a);
}
----

include::../see.adoc[]

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
