include::../description.adoc[]

== Noncompliant Code Example

With default provided regular expression ``^[A-Z][a-zA-Z0-9]*$``:

----
public class myClass {...}
----

== Compliant Solution

----
public class MyClass {...}
----
