include::../description.adoc[]

== Noncompliant Code Example

----
IF A = 0. " Noncompliant
ELSEIF A > 1. " Noncompliant 
ELSE. " Noncompliant
ENDIF.
----

include::../exceptions.adoc[]

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
