When allocating memory with ``++malloc++``, ``++calloc++`` and ``++realloc++`` it is important to make sure that the size of the allocated memory is compatible with the receiver type.


== Noncompliant Code Example

----
void f() {
  ...
  int* ptr1 = malloc(sizeof(short)); // Noncompliant
  ...
}
----


== Compliant Solution

----
void f() {
  ...
  int* ptr1 = malloc(sizeof(int));
  ...
}
----

