It is possible to use the increment operator ``++``, to set the value of a ``bool``(C++) or ``_Bool``(C) variable to ``true``. But this feature has been deprecated in C++ since the 1998 version of the standard, removed in C++17, and even where allowed, is simply confusing. 


== Noncompliant Code Example

----
bool alive;
...
alive++;
----


== Compliant Solution

----
bool alive;
...
alive = true;
----


== See

*  ISO/IEC 14882:1998, 5.3.2

