include::../description.adoc[]

== Noncompliant Code Example

----
DECLARE @a INT;
DECLARE @b INT = 2;
SET @a = @a; -- Noncompliant
----

== Compliant Solution

----
DECLARE @a INT;
DECLARE @b INT = 2;
SET @a = @b;
----

include::../see.adoc[]

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
