include::../description.adoc[]

== Noncompliant Code Example

----
If BooleanVariable = True Then ...
If BooleanVariable <> True Then ...
If BooleanVariable OR False Then ...
DoSomething(Not False)
----

== Compliant Solution

----
If BooleanVariable Then ...
If Not BooleanVariable Then ...
If BooleanVariable Then ...
DoSomething(True)
----
