include::../description.adoc[]

== Noncompliant Code Example

----
  if (true) {
    doSomething();
  }
  // ...
  if (false) {
    doSomethingElse();
  }
----

== Compliant Solution

----
  doSomething();
  // ...
----

include::../see.adoc[]

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
