include::../description.adoc[]

== Noncompliant Code Example

----
public bool CanVote(Person person) 
{
  return person.GetAge() > 18 ? true : true; // Noncompliant; is this what was intended?
}
----

== Compliant Solution

----
public bool CanVote(Person person) 
{
  return person.GetAge() > 18 ? true : false; 
  // or even better: 
  // return person.GetAge() > 18;
}
----

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
