include::../description.adoc[]

== Noncompliant Code Example

With the default regular expression ``++^[a-z][a-zA-Z0-9]*$++``:

----
public function doSomething($my_param){  
  $LOCAL;
  ...
}
----

== Compliant Solution

----
public function doSomething($myParam){  
  $local;
  ...
}
----
