include::../description.adoc[]

== Noncompliant Code Example

With the default regular expression ``++^[a-z][a-zA-Z0-9]*$++``:

----
public function doSomething($my_param){  
  $LOCAL;
  ...
}
----

== Compliant Solution

----
public function doSomething($myParam){  
  $local;
  ...
}
----

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
