include::../description.adoc[]

== Noncompliant Code Example

With the default regular expression ``++^[a-z_]+(\.[a-z_][a-z0-9_]*)*$++``:

----
package org.exAmple; // Noncompliant
----

== Compliant Solution

----
package org.example; 
----

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
