Shared naming conventions allow teams to  collaborate efficiently. This rule checks that static non-final field names match a provided regular expression.


== Noncompliant Code Example

With the default regular expression ``++^[a-z][a-zA-Z0-9]*$++``:

----
public final class MyClass {
   private static String foo_bar;
}
----


== Compliant Solution

----
class MyClass {
   private static String fooBar;
}
----


