include::../description.adoc[]

== Noncompliant Code Example

----
if (condition1) {
  if (condition2) {        
    ...
  }
}
----

== Compliant Solution

----
if (condition1 && condition2) {        
  ...
}
----

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
