For better readability, it is preferable to use `+DECLARE+` instead of the unpronounceable `+DCL+` abbreviation.


== Noncompliant Code Example

----
foo: proc options(main);
  dcl i fixed decimal init (0); /* Noncompliant */
end;
----


== Compliant Solution

----
foo: proc options(main);
  declare i fixed decimal init (0); /* Compliant */
end;
----

