include::../description.adoc[]

== Noncompliant Code Example

----
if (!(a === 2)) { ... }  // Noncompliant
----

== Compliant Solution

----
if (a !== 2) { ... } 
----

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
