include::../description.adoc[]

== Noncompliant Code Example

----
Sub DoSomething() 
    ' TODO
End Sub
----

include::../see.adoc[]

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
