=== On 2015-07-09T15:46:00Z Ann Campbell Wrote:
\[~tamas.vajk] I know that as written this rule doesn't cover _BitwiseOperatorOnEnumWithoutFlags_ but IMO it obviates it. According to http://stackoverflow.com/a/8480/2662707[SO] the ``++[Flags]++`` doesn't actually get you anything w/r/t bitwise usage. What's important is the assigned values.


I propose a Minor followup rule to make sure that ``++[Flags]++`` is applied when the enum values are power-of-2. 


WDYT?

=== On 2015-07-10T09:41:50Z Tamas Vajk Wrote:
\[~ann.campbell.2] I've modified slightly the messages and the sample code. I think now it covers your proposed follow-up rule as well.

=== On 2015-07-10T11:14:07Z Ann Campbell Wrote:
Thanks [~tamas.vajk]

=== On 2015-09-01T07:02:18Z Dinesh Bolkensteyn Wrote:
IMO it should be another rule (a MINOR one) that suggests to mark power-of-two Enums with ``++Flags++`` - that knowledge cannot just sneak into this rule's issue message [~ann.campbell.2].

=== On 2015-09-01T14:21:41Z Ann Campbell Wrote:
That's between you [~tamas.vajk], [~dinesh.bolkensteyn].

=== On 2015-09-02T12:20:36Z Tamas Vajk Wrote:
\[~ann.campbell.2] We discussed it with [~dinesh.bolkensteyn] and he convinced me that we should have a minor severity rule that proposes to use the ``++[Flags]++`` whenever an enum only contains power-of-two items. (We can't merge that one into this issue, because of the different severity, but has a value, because it improves the understability of the code)

=== On 2015-09-02T15:22:59Z Tamas Vajk Wrote:
\[~ann.campbell.2] We realized during the implementation that this power-of-two enum value restriction can't be enforced, because 

----
enum Permission
{ 
  None = 0,
  Read = 1,   
  Write = 2, 
  ReadWrite = Read  | Write, //means 3
  Execute = 4 
}
----
is totally valid. So there is no need for the other RSPEC which we discussed above. I'll change this one.

=== On 2015-09-02T15:24:59Z Tamas Vajk Wrote:
\[~dinesh.bolkensteyn] could you please review this rspec? Thanks

=== On 2015-10-19T12:55:44Z Tamas Vajk Wrote:
This RSPEC is closed, as we realized during the implementation that this power-of-two enum value restriction can't be enforced.

=== On 2015-12-03T09:32:15Z Tamas Vajk Wrote:
\[~ann.campbell.2] I reopened this RSPEC. I'm going through the R# rules, and I think it would still make sense. Note that the rule was closed because we realized that flags enums can't be forced to have only power of two values. So I removed that part, and only left the bitwise operation part in the rule. WDYT?

=== On 2015-12-04T18:40:13Z Ann Campbell Wrote:
\[~tamas.vajk] I moved the Noncompliant marker in the Noncompliant Code Example.

=== On 2015-12-07T15:59:08Z Tamas Vajk Wrote:
\[~ann.campbell.2] I see the point of marking the declaration noncompliant, but from a user perspective it would be better on the call:

Let's say we mark the `enum` noncompliant, and say that it should not be used as flags, if it's not marked with `[Flags]`. But the developer would need some verification that indeed it's used somewhere in a binary or/and. So we would probably include a filename and line number in the issue message. And then the dev could go check it there, and then come back to the declaration and fix it.


While in the other case if the usage is marked, the dev would only need to go to the definition, which is just one shortcut away.


What do you think?

=== On 2015-12-07T16:17:37Z Ann Campbell Wrote:
If we're going to do that [~tamas.vajk], then we need to flip the rule around: "Non-flags enums should not be used in bitwise operations" (note double-negative).

