include::../description.adoc[]

== Noncompliant Code Example

----
System.out.println("My Message");  // Noncompliant
----

== Compliant Solution

----
logger.log("My Message");
----

include::../see.adoc[]

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
