include::../description.adoc[]

include::../recommended.adoc[]

== Noncompliant Code Example

Below, the hashed password is not salted:

----
MessageDigest md = MessageDigest.getInstance("SHA-512");
byte[] hashedPassword = md.digest(passwordToHash.getBytes(StandardCharsets.UTF_8)); // Noncompliant
----

== Compliant Solution

Use ``++java.security.SecureRandom++`` to produce a unpredictable salt:

----
MessageDigest md = MessageDigest.getInstance("SHA-512");

SecureRandom random = new SecureRandom();
byte[] salt = new byte[16];
random.nextBytes(salt);

md.update(salt);

byte[] hashedPassword = md.digest(passwordToHash.getBytes(StandardCharsets.UTF_8)); // Compliant
----

include::../see.adoc[]
