include::../description.adoc[]

== Noncompliant Code Example

----
public void doSomething() {
}
----

== Compliant Solution

----
public void doSomething() {
  // Do nothing because of X and Y.
}
----

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
